presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSinkProvider.java

```diff
@@ -82,7 +82,7 @@ public ConnectorPageSink createPageSink(ConnectorTransactionHandle transactionHa
 
         RaptorInsertTableHandle handle = (RaptorInsertTableHandle) tableHandle;
         return new RaptorPageSink(
-                new FileSystemContext(session),
+                new HdfsContext(session.getIdentity()),
```

Comment:
Could you address the comments https://github.com/prestodb/presto/pull/14680/files#r443993637? We need more than `identity`.

Reply:
Should I populate the fields with an empty string?

Reply:
Create a new interface `public HdfsContext(ConnectorSession session)` for `HdfsContext`

Reply:
BTW, the constructors for HdfsContext is really bad. We should have something like this
```java
    public HdfsContext(ConnectorSession session)
    {
        this(session, Optional.empty(), Optional.empty());
    }

    public HdfsContext(ConnectorSession session, String schemaName)
    {
        this(session, Optional.of(schemaName), Optional.empty());
    }

    public HdfsContext(ConnectorSession session, String schemaName, String tableName)
    {
        this(session, Optional.of(schemaName), Optional.of(tableName));
    }

    private HdfsContext(ConnectorSession session, Optional<String> schemaName, Optional<String> tableName)
    {
        requireNonNull(session, "session is null");
        this.identity = requireNonNull(session.getIdentity(), "session.getIdentity() is null");
        this.source = requireNonNull(session.getSource(), "session.getSource() is null");
        this.queryId = Optional.of(session.getQueryId());
        this.schemaName = requireNonNull(schemaName, "schemaName is null");
        this.tableName = requireNonNull(tableName, "tableName is null");
        this.clientInfo = session.getClientInfo();
    }

Reply:
Ok! Doing an explicit constructor invocation makes it clearer.